### PR TITLE
Integrate the full-training flow into CoreEngine

### PR DESCRIPTION
**Pull request by BenCarter44**
This pull request covers the basic training script that can be run to extract and train the ML models.

The UI repository will then later use the saved models to make predictions.

### CODE REVIEW

**Comment by BenCarter44** [comment]
The tree-sitter submodule/dependency is no longer needed, as it can be all wrapped up in a pip install tree-sitter-java --> now in the pyproject.toml

File: main.py [hunk 1]
```diff
@@ -33,42 +47,96 @@ def main():
...
+    if method == "gpt":
+        system_message, assistant_message = classifier.generate_system_message(
+            sub_labels, df
+        )
+        classifier.generate_gpt_messages(system_message, assistant_message, df)
+
+        llm_classifier = classifier.fine_tune_gpt()
+
+        # Save Model
+        with open(cfg_dict["classification_model_save"], "wb") as f:
```
Comment (mcauley-penney) [inline main.py:71]: `classification_model_save` seems to be a path where a classification model will be output to. I had to look around to understand what it meant and I think a more fitting name could be chosen.

The most important part imo is that 'save' is typically read as a verb, not a noun that describes a location. Because of this, this initially seemed to me like the variable name for a boolean. Something like `classification_model_path` or `classification_model_output_path` would be better. 

If we want to be shorter, scikit-learn commonly uses `clf` as a shorthand for the concept of a classifier (whether this is the same type of classifier may or may not be important; see [here](https://scikit-learn.org/stable/tutorial/basic/tutorial.html#learning-and-predicting)). So, something like `clf_model_path`/`clf_model_out_path` might be clear to those in this niche of CS while also being shorter. The downside is that it might confuse anyone with sklearn experience.

```suggestion
        with open(cfg_dict["clf_model_out_path"], "wb") as f:
```

File: main.py [hunk 2]
```diff
@@ -33,42 +47,96 @@ def main():
...
 
-    #     data_from_extraction = database manager query OutputTable
-    #
-    #     rf_model = llm_classifier.train_rf(domain_dictionary, data_from_extraction)
-
-    #     save_model(rf_model)
+    db.save()
 
-    # exit()
+    method = cfg_dict["classification_method"]
```
Comment (mcauley-penney) [inline main.py:57]: The comment below could also be applied here, which is something I contributed. We could probably shorten this to

```suggestion
    method = cfg_dict["clf_method"]
```

File: src/repo_extractor/schema.py [hunk 3]
```diff
@@ -218,6 +218,8 @@ def _get_userlogin(api_obj) -> str:
     "output_path": _str_type,
     "api_domain_label_listing": _str_type,
     "api_subdomain_label_listing": _str_type,
+    "classification_method": _str_type,
+    "classification_model_save": _str_type,
```
Comment (mcauley-penney) [inline src/repo_extractor/schema.py:222]: We should add documentation to support these additions to the schema, including the label listings. The extractor already has documentation for its original schema, so we just need to add these four items, and the names of the classification keys should be changed to match whatever changes we make to them in the source code after this review.

**Review by mcauley-penney — CHANGES_REQUESTED** [review]
My comments are soft asks; if you disagree with the requested name changes, please discuss.

**Review by mcauley-penney — APPROVED** [review]
